stream-sdk/stream-parent/stream-dbs-web-client/src/main/java/com/backbase/stream/webclient/filter/HeadersForwardingServerFilter.java

```diff
@@ -0,0 +1,40 @@
+package com.backbase.stream.webclient.filter;
+
+import static com.backbase.stream.webclient.DbsWebClientConfiguration.CONTEXT_KEY_FORWARDED_HEADERS;
+import static java.util.Arrays.asList;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.server.RequestPath;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Component
+public class HeadersForwardingServerFilter implements WebFilter {
+
+    private static final List<String> HEADERS_TO_FORWARD = asList("X-TID");
```

Comment:
It will be better if the forwarded headers are configured externally, so no code change will be required if more headers are needed in the future.

Reply:
agreed

Reply:
fixed